crates/oxc_semantic/src/builder.rs

```diff
@@ -453,6 +472,15 @@ impl<'a> SemanticBuilder<'a> {
                 self.function_stack.push(self.current_node_id);
                 func.bind(self);
                 self.make_all_namespaces_valuelike();
+                if let Some(body) = &func.body {
```

Comment:
We already track variable bindings. All of this code seems redundant. Additionally, you are traversing the AST in the "pre node" function, when this logic belongs within visit_`<node kind>` methods.

Reply:
Agreed, as I commented here https://github.com/web-infra-dev/oxc/issues/520#issuecomment-1622026569

We just need to store the redeclared `var`s.

Reply:
Yeah, I have read the issue and related code as @Boshen mentioned. But I occurred the case: 
```js
class C { static { var a; { var a; } } }
```
`oxc_semantic` produce difference `scope_id`, which is 1 and the `a` in block is `2`. 

So I re-checked variable declaration again, or if other else code or information I missed plz let me know.

Reply:
🤔 The code block I mentioned walks up the scope and checks for declaration of `var`, we need to save the redeclared information in the `Some` case, what context are we missing here?

Reply:
Thanks for your guidances. I refactored the code.